### ISSUE TITLE
Core modules to remove

### ISSUE THREAD

**Issue by swelljoe** [webmin/webmin#1089]
In the interest of deprecating some modules that should, for one or more reasons, be removed from the core Webmin distribution and the reasons I rhink they should be removed:

  - burner - It's a CD burner. It's 2019.
  - ppp-client - It's a dialup client and it's unmaintained upstream AFAICT.
  - vgetty - Upstream seems to have been unmaintained for more than a dozen years (unless I just failed to find the real upstream). Asterisk is seemingly the industry standard for voice services.
  - mon - Unmaintained for at least a dozen years
  - heartbeat - Replaced by pacemaker, which is wholly incompatible and does not have a tool named heartbeat, any more. The last heartbeat release was ~15 years ago.
  - jabberd - As far as I can tell, jabberd has been superceded by the incompatible jabberd2 more than a decade ago. It might  be possible to update the module to support jabberd, but no one has sent patches, so I assume it won't happen.
  - wuftpd - Unmaintained for several years. Development group websites are all down, download servers are, too. Last findable release was many years ago, and it has been removed from all distributions I pay attention to for several years due to security issues and no maintenance.

**Comment by Neustradamus** [comment]
jabberd2 is not a good solution, the development has been stopped.

There are several actual XMPP servers, in order:
- ejabberd: Warning, only support SCRAM-SHA-1
- Jackal: SCRAM-SHA-1(-PLUS), SCRAM-SHA-256(-PLUS), SCRAM-SHA-512(-PLUS)
- Metronome IM: SCRAM-SHA-1(-PLUS), SCRAM-SHA-256(-PLUS), SCRAM-SHA-512(-PLUS)
- Prosody IM: Warning, only supports SCRAM-SHA-1(-PLUS)
- Tigase: SCRAM-SHA-1(-PLUS), SCRAM-SHA-256(-PLUS), SCRAM-SHA-512(-PLUS)

**Comment by swelljoe** [comment]
We don't have any plans to support jabberd2, I just wanted to document what happened to jabberd and why it needs to be removed. If someone contributes a module for those other options, we can consider adding something.

**Comment by shoulders** [comment]
@swelljoe Did these get removed? Less is more.

**Comment by iliajie** [comment]
> * Prosody IM: Warning, only supports SCRAM-SHA-1(-PLUS)

I'm interested in creating a module for Prosody. Though, it can only be a side project.

> Did these get removed? Less is more.

Currently only `heartbeat` and `ppp-client` are in standard package:

https://github.com/webmin/webmin/blob/master/mod_def_list.txt

**Comment by Neustradamus** [comment]
@iliaross: About XMPP, have you progressed?

**Comment by iliaross** [comment]
> [@iliaross](https://github.com/iliaross): About XMPP, have you progressed?

No, I never started it yet. But I want to do it, most likely after the Virtualmin 8 release...

**Comment by shoulders** [comment]
It would make more sense for the plugins To be removed in version 8 seeing as it is a major version. Obviously I mean they are not present in the first version of 8.

**Comment by Neustradamus** [comment]
@iliaross: Thanks for your answer!
You can see for ejabberd and/or Prosody IM.

**Comment by iliaross** [comment]
> No, I never started it yet. But I want to do it, most likely after the Virtualmin 8 release..

I think it's a great time to do it! I really like Prosody, and many people can benefit from using it today.

**Comment by Neustradamus** [comment]
@swelljoe: It has been closed but it has been solved?

Important: ejabberd has very changed since my comment from 2019 about security (better SCRAM security).
- https://github.com/processone/ejabberd/releases

cc: @mremond.

**Comment by iliaross** [comment]
> [@swelljoe](https://github.com/swelljoe): It has been closed but it has been solved?

If you mean creating a Prosody module, not yet, but I’m working on it now. And, you’re welcome to open a new ticket for any other issue mentioned here that you think is still unresolved.
